src/cmd-build

```diff
@@ -172,16 +176,11 @@ touch .build-commit
 mv -T ${tmp_builddir} ${buildid}
 # Replace the latest link
 ln -Tsfr "${buildid}" latest
-python3 -c 'import os,json,sys,stat;
-builds=[]
-for d in os.listdir("."):
-  s = os.lstat(d)
-  if not (stat.S_ISDIR(s.st_mode) and os.path.isfile(d + "/meta.json")):
-    continue
-  builds.append((d, s.st_ctime))
-json.dump({"builds":
-           list(map(lambda x: x[0], sorted(builds, key=lambda x: x[0])))},
-          sys.stdout)
-' > ${workdir}/tmp/builds.json
-mv ${workdir}/tmp/builds.json builds.json
+# And now clean up previous builds and regen builds.json
+KEEP_LAST_N=
+if [ ${SKIP_PRUNE} = 1 ]; then
+    # Really then, we're just using the `prune_builds` to regen builds.json
```

Comment:
We *could*, though is it really worth it? Note that when we walk the `builds/` dir in `prune_builds`, we're also using the results of that to determine which builds need to be deleted. Splitting it up means we'd have duplicate code for walking and sorting the builds.

Reply:
I think once we rewrite we would just be able to share the code (i.e. shared python function). The other thing you can do is not "walk" the structure in `prune_builds` but just read the json (generated before prune) and delete the directories, then regenerate builds.json again after prune. Then we wouldn't be duplicating code as much. 

> We could, though is it really worth it?

I'll leave the above suggestion up to you. I was just mostly trying to think of how we could do things more cleanly.

Reply:
I started hacking on this like this:

- Factor out regen to `cmdlib` function
- Make builds call regen, and then if not skipping prune, call `cmd-prune`.
- Change `cmd-prune` to also use the regen function.

But one thing I really don't like is that now we're regenerating a total of three times:

- once after `build`
- once in `cmd-prune` before pruning; i.e. we don't want to rely on an accurate `builds.json`, otherwise, we might miss out on dirs that somehow aren't in there.
- once more after pruning is over.

So overall, I think I still prefer the current approach, though I appreciate your apprehension of the `--keep-last-n=0` approach here. I can add an explicit `--just-regen` if you'd prefer?

Reply:
> I can add an explicit `--just-regen` if you'd prefer?

nah, seems fine. I think all of this will change soon enough anyway. Maybe just add a note to the code to factor our the regen in the future to a common python library.

Reply:
I added a note to `prune_builds`! :arrow_down: